[Bug] Failed to build clucene by GCC 12.2

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

Doris version: master
OS: Centos 6
Compiler: GCC 12.2

### What's Wrong?

```shell
[20/212] Original Turbo-PFOR makefile target
FAILED: CMakeFiles/build_ic /Programs/doris/thirdparty/src/doris-thirdparty-libclucene-v2.4.8/doris_build/CMakeFiles/build_ic
cd /Programs/doris/thirdparty/src/doris-thirdparty-libclucene-v2.4.8/src/ext/for && make OPT='-Wno-implicit-function-declaration -Wno-implicit-int -Wno-int-conversion' USE_AVX2=1 libic.a -j 8
/devel/compiler/bin/../bin/gcc -O3 -w -mavx2 -DAVX2_ON -Wno-implicit-function-declaration -Wno-implicit-int -Wno-int-conversion -c bitpack.c -o bitpack_avx2.o
/devel/compiler/bin/../bin/gcc -O3 -w -mavx2 -DAVX2_ON -Wno-implicit-function-declaration -Wno-implicit-int -Wno-int-conversion -c bitunpack.c -o bitunpack_avx2.o
/devel/compiler/bin/../bin/gcc -O3 -w -mavx2 -DAVX2_ON -Wno-implicit-function-declaration -Wno-implicit-int -Wno-int-conversion -c vp4c.c -o vp4c_avx2.o
/devel/compiler/bin/../bin/gcc -O3 -w -mavx2 -DAVX2_ON -Wno-implicit-function-declaration -Wno-implicit-int -Wno-int-conversion -c vp4d.c -o vp4d_avx2.o
/devel/compiler/bin/../bin/gcc -O3 -w -mavx2 -DAVX2_ON -Wno-implicit-function-declaration -Wno-implicit-int -Wno-int-conversion -c transpose.c -o transpose_avx2.o
In file included from bitpack.c:30:
bitutil.h:192:31: error: conflicting types for '_mm_cvtsi128_si16'; have 'uint16_t(__m128i)' {aka 'short unsigned int(__m128i)'}
  192 | static ALWAYS_INLINE uint16_t _mm_cvtsi128_si16(__m128i v) { return (uint16_t)_mm_cvtsi128_si32(v); }
      |                               ^~~~~~~~~~~~~~~~~
In file included from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/immintrin.h:98,
                 from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/x86intrin.h:32,
                 from conf.h:232,
                 from bitpack.c:29:
/devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/avx512fp16intrin.h:2746:1: note: previous definition of '_mm_cvtsi128_si16' with type 'short int(__m128i)'
 2746 | _mm_cvtsi128_si16 (__m128i __A)
      | ^~~~~~~~~~~~~~~~~
In file included from vp4d.c:36:
bitutil.h:192:31: error: conflicting types for '_mm_cvtsi128_si16'; have 'uint16_t(__m128i)' {aka 'short unsigned int(__m128i)'}
  192 | static ALWAYS_INLINE uint16_t _mm_cvtsi128_si16(__m128i v) { return (uint16_t)_mm_cvtsi128_si32(v); }
      |                               ^~~~~~~~~~~~~~~~~
In file included from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/immintrin.h:98,
                 from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/x86intrin.h:32,
                 from conf.h:232,
                 from vp4d.c:35:
/devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/avx512fp16intrin.h:2746:1: note: previous definition of '_mm_cvtsi128_si16' with type 'short int(__m128i)'
 2746 | _mm_cvtsi128_si16 (__m128i __A)
      | ^~~~~~~~~~~~~~~~~
In file included from bitunpack.c:28:
bitutil.h:192:31: error: conflicting types for '_mm_cvtsi128_si16'; have 'uint16_t(__m128i)' {aka 'short unsigned int(__m128i)'}
  192 | static ALWAYS_INLINE uint16_t _mm_cvtsi128_si16(__m128i v) { return (uint16_t)_mm_cvtsi128_si32(v); }
      |                               ^~~~~~~~~~~~~~~~~
In file included from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/immintrin.h:98,
                 from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/x86intrin.h:32,
                 from conf.h:232,
                 from bitunpack.c:27:
/devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/avx512fp16intrin.h:2746:1: note: previous definition of '_mm_cvtsi128_si16' with type 'short int(__m128i)'
 2746 | _mm_cvtsi128_si16 (__m128i __A)
      | ^~~~~~~~~~~~~~~~~
In file included from vp4c.c:36:
bitutil.h:192:31: error: conflicting types for '_mm_cvtsi128_si16'; have 'uint16_t(__m128i)' {aka 'short unsigned int(__m128i)'}
  192 | static ALWAYS_INLINE uint16_t _mm_cvtsi128_si16(__m128i v) { return (uint16_t)_mm_cvtsi128_si32(v); }
      |                               ^~~~~~~~~~~~~~~~~
In file included from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/immintrin.h:98,
                 from /devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/x86intrin.h:32,
                 from conf.h:232,
                 from vp4c.c:33:
/devel/compiler/lib/gcc/x86_64-linux-gnu/12.2.0/include/avx512fp16intrin.h:2746:1: note: previous definition of '_mm_cvtsi128_si16' with type 'short int(__m128i)'
 2746 | _mm_cvtsi128_si16 (__m128i __A)
      | ^~~~~~~~~~~~~~~~~
make: *** [makefile:96: vp4d_avx2.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [makefile:90: vp4c_avx2.o] Error 1
make: *** [makefile:102: bitpack_avx2.o] Error 1
make: *** [makefile:108: bitunpack_avx2.o] Error 1
```

### What You Expected?

Build the third party libraries successfully.

### How to Reproduce?

```
./build-thirdparty.sh
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/thirdparty/CHANGELOG.md
+++ b/thirdparty/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 This file contains version of the third-party dependency libraries in the build-env image. The docker build-env image is apache/doris, and the tag is `build-env-${version}`
 
+## v20230411
+- Modified: clucene 2.4.8 -> 2.4.9
+
 ## v20230328
 - Modified: brpc 1.2.0 -> 1.4.0
 - Modified: boost 1.73.0 -> 1.81.0

--- a/thirdparty/vars.sh
+++ b/thirdparty/vars.sh
@@ -55,10 +55,10 @@ export TP_JAR_DIR="${TP_INSTALL_DIR}/lib/jar"
 #####################################################
 
 #clucene
-CLUCENE_DOWNLOAD="https://github.com/apache/doris-thirdparty/archive/refs/tags/libclucene-v2.4.8.tar.gz"
-CLUCENE_NAME="doris-thirdparty-libclucene-v2.4.8.tar.gz"
-CLUCENE_SOURCE="doris-thirdparty-libclucene-v2.4.8"
-CLUCENE_MD5SUM="4d77af3c2c0d3c47754ffa68c277b9d6"
+CLUCENE_DOWNLOAD="https://github.com/apache/doris-thirdparty/archive/refs/tags/libclucene-v2.4.9.tar.gz"
+CLUCENE_NAME="doris-thirdparty-libclucene-v2.4.9.tar.gz"
+CLUCENE_SOURCE="doris-thirdparty-libclucene-v2.4.9"
+CLUCENE_MD5SUM="2d5c48bd24b0757d0fecb74111b9b2ed"
 
 # libevent
 LIBEVENT_DOWNLOAD="https://github.com/libevent/libevent/archive/release-2.1.12-stable.tar.gz"

